packages/java/endpoint/src/main/java/com/vaadin/hilla/route/RouteUnifyingIndexHtmlRequestListener.java

```diff
@@ -207,4 +253,37 @@ private Map<String, RouteParamType> getRouteParameters(
         });
         return routeParameters;
     }
+
+    boolean isClientMenuUsed(VaadinService vaadinService) {
+        try {
+            String fileRoutesJson = readFileRoutesJsonFile(
+                    ApplicationConfiguration.get(vaadinService.getContext()));
+            var json = mapper.readTree(fileRoutesJson);
+            // as the client registry has no information about layouts, the JSON
+            // is parsed again to search for a root node with children, which
+            // means that we have a root layout.
+            return json.has(0) && json.get(0).has("children")
+                    && json.get(0).get("children").isArray()
+                    && !json.get(0).get("children").isEmpty();
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    private static String readFileRoutesJsonFile(
+            AbstractConfiguration configuration) throws IOException {
+        if (configuration.isProductionMode()) {
+            try (InputStream inputStream = FrontendUtils.class
+                    .getResourceAsStream(FILE_ROUTES_JSON_PROD_PATH)) {
+                if (inputStream != null) {
+                    return IOUtils.toString(inputStream, UTF_8);
+                }
+                return "";
+            }
+        }
+        return FileUtils.readFileToString(new File(
+                FrontendUtils.getFrontendGeneratedFolder(
+                        FrontendUtils.getProjectFrontendDir(configuration)),
+                FILE_ROUTES_JSON_NAME), UTF_8);
+    }
```

Comment:
This logic to decide where should we look for `file-routes.json` in different modes reminds me of the `ClientRouteRegistry::getViewsJsonAsResource` (which is currently private). I suggest not to redefine the `FILE_ROUTES_JSON_NAME` and `FILE_ROUTES_JSON_PROD_PATH` duplicate constants and logic here and reuse the method from `ClientRouteRegistry` for getting it as a resource.

Reply:
It also makes sense to move the newly introduced method `isClientMenuUsed` to `clientRouteRegistry` and let it be the only place that has to deal with the logic of where to load the `file-routes.json`.

Reply:
Makes sense. I'll refactor it a bit.

Reply:
Done, also added couple tests.